index.js

```diff
@@ -17,7 +17,9 @@ module.exports.pitch = function(request) {
 	var callback = this.async();
 
 	var options = loaderUtils.getOptions(this) || {};
-	var filename = options.name || path.basename(this.resourcePath);
+	var context = options.context || this.rootContext;
+	var name = options.name || path.basename(this.resourcePath);
+	var filename = loaderUtils.interpolateName(this, name, { context: context });
```

Comment:
I'm unsure about this change; I used to use `interpolateName` but I removed it in v6 (https://github.com/erikdesjardins/spawn-loader/compare/v5.0.0...v6.0.0#diff-168726dbe96b3ce427e7fedce31bb0bcL25).

Although it's been so long that I don't quite remember why...in the release notes I said "...`[hash]` is correctly updated when subresources change..." but this PR didn't break the hashes in the tests, so maybe it's fine? OTOH maybe I meant "when subresources change during watch builds" which isn't tested.

Thoughts?

Reply:
@erikdesjardins Haven't noticed any problems in watch mode with my PR.

The reason I re-added `interpolateName` (and updated `loader-utils`) is because this allows for `options.name` to be a function returning a placeholder string. Passing this function as `filename` option directly to `this._compilation.createChildCompiler` doesn't work, so that's why I'm evaluating it before passing it to the child compiler.